src/utils/events/index.ts

```diff
@@ -3,63 +3,173 @@ import {
   AbiEnums,
   AbiEvents,
   AbiStructs,
-  Cairo1Event,
+  Cairo1Events,
   EventAbi,
   LegacyEvent,
   ParsedEvent,
   ParsedEvents,
   RPC,
+  type Cairo1Event,
+  type Cairo1EventVariant,
 } from '../../types';
+import assert from '../assert';
+import { isCairo1Abi } from '../calldata/cairo';
 import responseParser from '../calldata/responseParser';
 import { starkCurve } from '../ec';
 import { addHexPrefix, utf8ToArray } from '../encode';
 
 /**
  * Retrieves the events from the given ABI.
  *
+ * Is able to handle Cairo 1 events nested in Cairo components.
  * @param {Abi} abi - The ABI to extract events from.
  * @return {AbiEvents} - An object containing the extracted events.
+ * @example
+ * ```typescript
+ * const result = events.getAbiEvents(abi);
+ * // result = {
+ * //   '0x22ea134d4126804c60797e633195f8c9aa5fd6d1567e299f4961d0e96f373ee': 
+ * //    { '0x34e55c1cd55f1338241b50d352f0e91c7e4ffad0e4271d64eb347589ebdfd16': {
+ * //     kind: 'struct', type: 'event',
+ * //     name: 'ka::ExComponent::ex_logic_component::Mint',
+      
+ * //     members: [{
+ * //      name: 'spender',
+ * //      type: 'core::starknet::contract_address::ContractAddress',
+ * //      kind: 'key'},
+ * //      { name: 'value', type: 'core::integer::u256', kind: 'data' }]},
+ * // ...
+ * ```
  */
 export function getAbiEvents(abi: Abi): AbiEvents {
-  return abi
-    .filter((abiEntry) => abiEntry.type === 'event' && (abiEntry.size || abiEntry.kind !== 'enum'))
-    .reduce((acc, abiEntry) => {
-      const entryName = abiEntry.name.slice(abiEntry.name.lastIndexOf(':') + 1);
-      const abiEntryMod = { ...abiEntry };
-      abiEntryMod.name = entryName;
-      return {
-        ...acc,
-        [addHexPrefix(starkCurve.keccak(utf8ToArray(entryName)).toString(16))]: abiEntryMod,
-      };
-    }, {});
+  let finalResult: AbiEvents;
+  if (!isCairo1Abi(abi)) {
```

Comment:
are we testing now both of these cases in our tests, Cairo 0 and Cairo 1? 
We could even extract this logic in `if` and `else` Cairo 1 abi logic into 2 functions and do independent test for both of those new functions. Also, this function would become smaller if we do it this way

Reply:
Cairo 0 events have never been tested in the test suite. I have tested locally (with success) with an old customized Cairo 0 account that emits Cairo 0 events. If you want, I can add it in the test suite, but it will take some time (declare, deploy, invoke). What do you prefer?

Reply:
As cairo0 is in support mode for existing deployed contracts as so we can switch also to mode kinda supporting it but should focus on new cairo only

Reply:
ok then, no need for a test here!